File: 128. Longest Consecutive Sequence/sol.go
package main

func longestConsecutive(nums []int) int {
	s := make(map[int]struct{})
	for _, n := range nums {
		s[n] = struct{}{}
	}

	starts := make([]int, 0)
	for key, _ := range s {
		if _, exists := s[key-1]; !exists {
			starts = append(starts, key)
		}
	}

	res := 0
	for _, st := range starts {
		curr := st
		currCount := 0
		exists := true
		for exists {
			currCount++
			curr = curr + 1
			_, exists = s[curr]
		}
		if currCount > res {
			res = currCount
		}

	}

	return res

}

File: 128. Longest Consecutive Sequence/sol.py
from typing import Optional
class ListNode:
    def __init__(self, nxt: Optional[ListNode] = None):
       self.nxt = nxt

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set()
        for n in nums:
            s.add(n)

        starts = []
        for n in s:
            if n - 1 not in s:
                starts.append(n)
        res = 0
        for n in starts:
            curr = n
            currCount = 1
            while curr + 1 in s:
                curr = curr + 1
                currCount += 1
            if currCount > res:
                res = currCount


        return res